builder/proxmox/common/config.go

```diff
@@ -636,12 +666,113 @@ func (c *Config) Prepare(upper interface{}, raws ...interface{}) ([]string, []st
 		log.Printf("OS not set, using default 'other'")
 		c.OS = "other"
 	}
-	ideCount := 0
-	sataCount := 0
-	scsiCount := 0
-	virtIOCount := 0
+	// validate iso devices
+	for idx := range c.ISOs {
```

Comment:
Doesn't need to happen in this PR, but I wonder if we can make each `ISOs` here have its own `Prepare` function with a similar signature as the global `Prepare`, if only to split the logic into something more bite-sized.